crates/solver/src/settlement_post_processing/optimize_score.rs

```diff
@@ -1,56 +1,37 @@
 use {
     super::SettlementSimulating,
-    crate::{
-        objective_value::Inputs,
-        settlement::Settlement,
-        solver::score_computation::ScoreCalculator,
-    },
-    anyhow::{anyhow, Result},
+    crate::{settlement::Settlement, solver::risk_computation::RiskCalculator},
+    anyhow::Result,
     ethcontract::Address,
     gas_estimation::GasPrice1559,
-    num::{BigRational, FromPrimitive},
-    shared::{external_prices::ExternalPrices, http_solver::model::Score},
 };
 
-pub async fn compute_score(
+pub async fn compute_success_probability(
     settlement: &Settlement,
     settlement_simulator: &impl SettlementSimulating,
-    score_calculator: &ScoreCalculator,
+    risk_calculator: &RiskCalculator,
     gas_price: GasPrice1559,
-    prices: &ExternalPrices,
     solver: &Address,
-) -> Result<Score> {
+) -> Result<f64> {
     let gas_amount = settlement_simulator
         .estimate_gas(settlement.clone())
         .await
         .map(|gas_amount| {
             // Multiply by 0.9 to get more realistic gas amount.
             // This is because the gas estimation is not accurate enough and does not take
             // the EVM gas refund into account.
-            gas_amount * 9 / 10
+            gas_amount.to_f64_lossy() * 0.9
```

Comment:
Yes, tenderly should return exact gas usage, but we never used it for simulation. Although it should not be too difficult to change this since we already use tenderly simulation `TenderlyApi` for access lists monitoring. The question is whether we want to use tenderly or not.